### PR TITLE
[10.0]crm_phonecall: Fix action to redirect to the new phonecall when its created from phonecall2phonecall wizard.

### PR DESCRIPTION
**Pull request by jesusVMayor**
Also improves the schedule_another_phonecall function to return a recordset with the new phonecalls instead of a dict of it.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please change module version number. You are also changing API returning something different, which should be avoided in an already published module. Isn't possible to do it with "old" method? Please put a TODO for changing it for next version instead.

**Comment by jesusVMayor** [comment]
fixed

File: crm_phonecall/wizard/crm_phonecall_to_phonecall.py [hunk 1]
```diff
@@ -56,16 +56,18 @@ def action_schedule(self):
...
                 this.name,
                 this.user_id.id,
                 this.team_id.id or False,
                 this.tag_ids.ids,
                 action=this.action,
             )
-
-        return phonecalls.redirect_phonecall_view()
+        new_phonecalls = self.env['crm.phonecall']
+        for new_phonecall in new_phonecalls_dict.values():
```
Comment (pedrobaeza) [inline crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: You can do `new_phonecall = reduce(lambda x, y: a + b, new_phonecalls_dict.values())` for reducing code

Comment (jesusVMayor) [reply crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: done

**Review by pedrobaeza — APPROVED** [review]
@Yajo @chienandalu please review

File: crm_phonecall/wizard/crm_phonecall_to_phonecall.py [hunk 2]
```diff
@@ -56,16 +56,17 @@ def action_schedule(self):
...
                 this.user_id.id,
                 this.team_id.id or False,
                 this.tag_ids.ids,
                 action=this.action,
             )
-
-        return phonecalls.redirect_phonecall_view()
+        new_phonecalls = reduce(lambda x, y: x + y,
+                                new_phonecalls_dict.values())
+        return new_phonecalls.redirect_phonecall_view()
```
Comment (yajo) [inline crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: If I'm not wrong, `new_phonecalls` is now a `list`, which shouldn't have a `redirect_phonecall_view()` method.

File: crm_phonecall/models/crm_phonecall.py [hunk 3]
```diff
@@ -139,6 +139,7 @@ def schedule_another_phonecall(self, schedule_time, call_summary,
                                    tag_ids=False, action='schedule'):
         """
         action :('schedule','Schedule a call'), ('log','Log a call')
+        TODO: return a recordset instead of a dict
```
Comment (yajo) [inline crm_phonecall/models/crm_phonecall.py]: I think the fix is much easier if you just do this. Simply add a param `return_recordset=False`, and return a recordset if it is `True`, to keep backwards compatibility. Then add to Roadmap a note about removing that feature and always returning a recordset, as a good migration reminder.

**Review by yajo — CHANGES_REQUESTED** [review]

File: crm_phonecall/wizard/crm_phonecall_to_phonecall.py (hunk 2 above)
Comment (pedrobaeza) [reply crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: No, you are wrong :stuck_out_tongue_winking_eye: 

It returns a recordset. The reduce is for that.

Comment (yajo) [reply crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: How does that happen? 👀

Comment (pedrobaeza) [reply crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: Well, check reduce documentation

Comment (yajo) [reply crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: OK this code was not quite clear to me, but now I understand. It wasn't the problem with `reduce` (I know how that works bro 😆), it's more about the actual implementation... well, read above comment.

In any case, I'd have made just `sum(new_phonecalls_dict.values())`

File: crm_phonecall/wizard/crm_phonecall_to_phonecall.py [hunk 4]
```diff
@@ -56,16 +56,17 @@ def action_schedule(self):
         phonecall_obj = self.env['crm.phonecall']
         phonecalls = phonecall_obj.browse(self._context.get('active_ids', []))
         for this in self:
-            phonecalls.schedule_another_phonecall(
+            new_phonecalls_dict = phonecalls.schedule_another_phonecall(
```
Comment (yajo) [inline crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: If I'm not wrong, you are here generating phone calls in an exponential fashion:

1. `phonecalls` contains N recordsets.
2. You iterate over `phonecalls` and call `schedule_another_phonecall()` on each of those with the current values, thus creating N * N phonecalls.
2. You only store the last call's result into `new_phonecalls_dict`, making it seem like only N phonecalls were created.

So either:

- Call `this.schedule_another_phonecall`
- Call only once (without `for` loop) `phonecalls.schedule_another_phonecall`
- I'm completely wrong and I don't understand what you're doing here

**Review by yajo — CHANGES_REQUESTED** [review]

File: crm_phonecall/wizard/crm_phonecall_to_phonecall.py (hunk 2 above)
Comment (pedrobaeza) [reply crm_phonecall/wizard/crm_phonecall_to_phonecall.py]: `sum` works with recordsets? TIL then

**Comment by jesusVMayor** [comment]
@Yajo fixed. I don't use the sum function, because it seems that it don't work with a list of recordset.

**Review by yajo — APPROVED** [review]

**Pull request merged**